testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/cluster/ClientInvalidationClusterTest.java

```diff
@@ -58,24 +59,38 @@ protected ClientRepresentation createEntity(ClientRepresentation client, Contain
 
     @Override
     protected ClientRepresentation readEntity(ClientRepresentation client, ContainerInfo node) {
-        ClientRepresentation u = null;
-        try {
-            u = entityResource(client, node).toRepresentation();
-        } catch (NotFoundException nfe) {
-            // expected when client doesn't exist
-        }
+        ClientRepresentation u = Retry.call(new Retry.Supplier<ClientRepresentation>() {
+            @Override
+            public ClientRepresentation get(int iteration) {
+                try {
+                    return entityResource(client, node).toRepresentation();
+                } catch (NotFoundException nfe) {
+                    return null;
+                }
+            }
+        }, 3, 5000);
```

Comment:
Not convinced about this change. Doesn't it suggest perhaps different behaviour on Quarkus?

Reply:
Could it be as simple as async vs sync replication in Infinispan?

Reply:
Yes, SYNC seems to be better but yet not 100% stable.

This retry is not different than the pauses that are already there to wait for the cluster to rebalance.

I can remove, but IMO it gives more stability.

Reply:
I see that. My point is that it is related to HA and as such, you usually have an LB even though you don't need it to actually check invalidation.

Please, let me know if you want to remove it here or in a separate PR.

Reply:
I think your change in this PR is fine.

The point I made is not directly related to your PR, but to the cluster invalidation tests in general - I think they don't need so much server restarts as they currently have. I would address this eventually in separate PR.

Reply:
Cool. @stianst I think we have now a blessing from @mposolda !